CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JMultiChoiceComboBoxMenuNGTest.java

```diff
@@ -0,0 +1,369 @@
+        instance.setText(text);
+        String result = instance.getText();
+        assertEquals(result, text);
+    }
+
+    /**
+     * Test of getIcon method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testGetIcon() {
+        System.out.println("getIcon");
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+        Icon result = instance.getIcon();
+        assertNotNull(result);
+    }
+
+    /**
+     * Test of setIcon method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testSetIcon() throws MalformedURLException {
+        System.out.println("setIcon");
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+
+        final Icon newIcon1 = new ImageIcon(Utilities.toURI(iconFile).toURL());
+        final Icon newIcon2 = new ImageIcon(DROP_DOWN_ARROW_ICON_PATH);
+        instance.setIcon(newIcon1);
+        Icon result = instance.getIcon();
+        assertEquals(newIcon1, result);
+
+        instance.setIcon(newIcon2);
+        result = instance.getIcon();
+        assertEquals(newIcon2, result);
+    }
+
+    /**
+     * Test of getItems method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testGetItems() {
+        System.out.println("getItems");
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+        Set result = instance.getItems();
+        assertEqualsNoOrder(result.toArray(), items.toArray());
+    }
+
+    /**
+     * Test of getSelectedItems method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testGetSelectedItems() {
+        System.out.println("getSelectedItems");
+        String item = "Item 3";
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+        instance.setSelectedItem(item);
+        Set resultSet = instance.getSelectedItems();
+        assertEquals(resultSet.size(), 1);
+        assertEquals(resultSet.stream().findFirst().get().toString(), item);
+    }
+
+    /**
+     * Test of setSelectedItem method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testSetSelectedItem() {
+        System.out.println("setSelectedItem");
+        String item = "Item 3";
+        JMultiChoiceComboBoxMenu instance = mock(JMultiChoiceComboBoxMenu.class);
+        instance.setSelectedItem(item);
+        verify(instance, times(1)).setSelectedItem(item);
+    }
+
+    /**
+     * Test of setSelectedItems method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testSetSelectedItems() {
+        System.out.println("setSelectedItems");
+        String s1 = "Item 1";
+        String s2 = "Item 3";
+        String s3 = "Item 4";
+        JMultiChoiceComboBoxMenu instance = mock(JMultiChoiceComboBoxMenu.class);
+        instance.setSelectedItems(s1, s2, s3);
```

Comment:
same here

Reply:
Because this is already used in another method, sonar now complaints of a 1.3% Duplication. Shall I remove this method?

Reply:
Above comment not relevant here, so I'll delete